File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/HeapEntity.java
package com.github.ol_loginov.heaplibweb.repository.heap;

import com.github.ol_loginov.heaplibweb.repository.EntityIdentity;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
public class HeapEntity extends EntityIdentity {
	private int fileId;
	private Instant tm = Instant.now();
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/HeapEntity.kt
package com.github.ol_loginov.heaplibweb.repository.heap

import com.github.ol_loginov.heaplibweb.repository.EntityIdentity
import com.github.ol_loginov.heaplibweb.repository.HeapFile
import java.time.ZonedDateTime

class HeapEntity private constructor(
    id: Int,
    var fileId: Int,
    var tm: ZonedDateTime,
    var tablePrefix: String
) : EntityIdentity(id) {
    constructor(file: HeapFile) : this(0, file.id, ZonedDateTime.now(), "")
    private constructor() : this(0, 0, ZonedDateTime.now(), "")

    fun generateTablePrefix() {
        this.tablePrefix = "Z${id}_"
    }
}